[LLDB] Switch to using DIL as default implementation for 'frame var'.

cmtice opened the pull request.

cmtice committed:
[LLDB] Switch to using DIL as default implementation for 'frame var'.

labath commented:
There are a couple of failures in the CI. The backtraces don't make a whole lot sense, but it looks like there's something wrong with retrieving the list of variables from a CU.

kuilpd commented:
I addressed `lldb-shell-subprocess` tests failing in #147955 , but there's still 2 data formatter tests failing, not sure what's happening there yet.

cmtice committed:
Merge remote-tracking branch 'origin/main' into DIL-switch

Michael137 commented:
> but there's still 2 data formatter tests failing, not sure what's happening there yet.

FWIW, the `object` child of a `std::shared_ptr` should be equivalent to dereferencing it. How does DIL handle the `$$dereference$$` operator? Does it do anything special?

cmtice commented:
> > but there's still 2 data formatter tests failing, not sure what's happening there yet.
> 
> FWIW, the `object` child of a `std::shared_ptr` should be equivalent to dereferencing it. How does DIL handle the `$$dereference$$` operator? Does it do anything special?

DIL handles it exactly the same way the current 'frame variable' handles it, namely it uses the provided synthetic children and data formatters. (Gets the synthetic value for the shared pointer, then calls 'GetChildMemberWithName("object")', which returns the correct thing.

cmtice committed:
Small fixes:
 - Remove not-always-valid field-name check when evaluating field members.
 - Fix unexpected passes in TestDAP_evaluate (test owner said it was ok).
 - Add more anonymous namespace tests.

cmtice committed:
Fix clang-format issues.

labath commented:
I gave this patch a spin on a mac. The only failure I got was in test/Shell/SymbolFile/DWARF/TestDedupWarnings.test, which fails it does not generate both of the warnings, and that's because the "p" (aka dwim-print) now resolves to "frame var" instead of expr". Changing the test to use "expr a/b" instead should fix it.

cmtice committed:
Update TestDedupWarnings.test to use 'expr' instead of 'p'.

labath reviewed (approved):
:crossed_fingers:

cmtice committed:
Merge remote-tracking branch 'origin/main' into DIL-switch

labath commented:
It looks like this just barely didn't make it into the 21.x release branch. Do we want to cherry-pick that over? I think it would be a good way to flush out any issues...

DavidSpickett commented:
Perhaps with a release note saying this happened and how to switch back to the non-default way?

DavidSpickett commented on llvm/docs/ReleaseNotes.md:
```diff
@@ -311,11 +311,20 @@ Changes to LLDB
+  [DIL](https://discourse.llvm.org/t/rfc-data-inspection-language/69893) as the
+  default implementation for 'frame variable'. This should not change the
+  behavior of 'frame variable' at all, at this time. To revert to using the
+  old implementation use
```
Add a `:` on the end here, or put the command on the same line, so it's one sentence.

DavidSpickett commented:
> This PR updates the release notes; is that not good enough?

Yes that's perfect, didn't see that it was already there.

kuilpd commented:
> It looks like this just barely didn't make it into the 21.x release branch. Do we want to cherry-pick that over? I think it would be a good way to flush out any issues...

I think so too, this and #149117. Do we just create a PR with the same changes into `release/21.x` branch?

labath commented:
The process for that is [here](https://llvm.org/docs/GitHub.html#backporting-fixes-to-the-release-branches), though it's a bit unfortunate that a week has passed since then (I was OOO this week so I couldn't reply sooner). At this point, I'm no longer sure it makes sense to do that.

kuilpd commented:
> The process for that is [here](https://llvm.org/docs/GitHub.html#backporting-fixes-to-the-release-branches), though it's a bit unfortunate that a week has passed since then (I was OOO this week so I couldn't reply sooner). At this point, I'm no longer sure it makes sense to do that.

We could still do it before the next minor release, at least I feel like it should be.

labath commented:
In that case, please create the cherry pick as soon as possible, and we can see where that discussion leads us. For myself, I'm going to defer @JDevlieghere opinion.

JDevlieghere commented:
> In that case, please create the cherry pick as soon as possible, and we can see where that discussion leads us. For myself, I'm going to defer @JDevlieghere opinion.

I'm supportive. For Swift we'll be [basing our release branch on the LLVM one](https://forums.swift.org/t/updating-llvm-project-branch-for-swift-main-stable-21-x/81184) and I'd like us to start living on it there as well.

labath commented:
/cherry-pick f5c676d6d95dc63b8dcb43dd2e4dfc913a9c5a50 33396d71e9aef624670380e11348a11b8ff74246
